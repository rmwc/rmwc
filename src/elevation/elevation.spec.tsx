import React from 'react';
import { render } from '@testing-library/react';
import { Elevation } from './';

describe('Elevation', () => {
  it('renders', () => {
    const { asFragment } = render(<Elevation z={0} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can have z value', () => {
    const { container } = render(<Elevation z="24" />);
    expect(container.firstChild).toHaveClass('mdc-elevation--z24');
  });

  it('can transition', () => {
    const { container } = render(<Elevation transition z={0} />);
    expect(container.firstChild).toHaveClass('mdc-elevation-transition');
  });

  it('can have custom classnames', () => {
    const { container } = render(
      <Elevation z={0} className={'my-custom-classname'} />
    );
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});
