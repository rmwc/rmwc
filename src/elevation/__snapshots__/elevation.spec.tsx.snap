// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elevation renders 1`] = `
<DocumentFragment>
  <div
    class="mdc-elevation--z0"
  />
</DocumentFragment>
`;
